core/assets/js/app.js

```diff
@@ -67,42 +69,74 @@ window.liveSocket = liveSocket
 
 
 // PWA
-if ('serviceWorker' in navigator) {
-  navigator.serviceWorker.register('./sw.js', {scope: './'})
-  .catch((error) => {
-    // registration failed
-    console.log('Registration failed with ' + error);
+//
+//
+const pushStore = Spruce.store("push", {registration: "pending"})
+const getExistingSubscription = () => {
+  return navigator.serviceWorker.ready.then((registration)=> {
+    return registration.pushManager.getSubscription().then(subscription=>{
+      return {registration, subscription};
+    })
   });
+}
+const registerPushSubscription = (subscription) => {
+      console.log("Server", subscription);
+  return fetch('/web-push/register', {
+    method: 'post',
+    headers: {
+      'Content-type': 'application/json'
+    },
+    body: JSON.stringify({
+      subscription: subscription
+    }),
+  }).then(()=>{
+    pushStore.registration = "registered"
+  });
+}
 
 
-  navigator.serviceWorker.ready.then((registration)=> {
-    return registration.pushManager.getSubscription().then((subscription)=> {
-
-      if (subscription) {
-        return subscription;
-      }
+window.registerForPush = ()=>{
+  if (!('serviceWorker' in navigator)) {
+    alert("Sorry, your browser does not support push")
```

Comment:
Do we need locale support here eventually?

Reply:
Yes, this needs to be moved over to LiveView.